solidity/contracts/converter/types/liquidity-pool-v2/LiquidityPoolV2Converter.sol

```diff
@@ -380,25 +336,31 @@ contract LiquidityPoolV2Converter is LiquidityPoolConverter {
         // validate input
         require(_sourceToken != _targetToken, "ERR_SAME_SOURCE_TARGET");
 
-        // get the external rate between the reserves along with its update time
-        Fraction memory externalRate;
-        uint256 externalRateUpdateTime;
-        (externalRate.n, externalRate.d, externalRateUpdateTime) =
-            priceOracle.latestRateAndUpdateTime(primaryReserveToken, secondaryReserveToken);
-
-        // get the source token effective / external weights
-        (uint32 sourceTokenWeight, uint32 externalSourceTokenWeight) = effectiveAndExternalPrimaryWeight(externalRate, externalRateUpdateTime);
-        if (_targetToken == primaryReserveToken) {
-            sourceTokenWeight = inverseWeight(sourceTokenWeight);
-            externalSourceTokenWeight = inverseWeight(externalSourceTokenWeight);
+        uint32 sourceTokenWeight;
+        uint32 targetTokenWeight;
+
+        // if the rate was already checked in this block, use the current weights; otherwise, get the new weights
+        if (externalRateUpdateTime == time()) {
+            sourceTokenWeight = reserves[_sourceToken].weight;
+            targetTokenWeight = reserves[_targetToken].weight;

```

Comment:
can use PPM_RESOLUTION - sourceTokenWeight here to save some gas

Reply:
Fixed in all places; please verify.